core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.kt

```diff
@@ -80,4 +87,53 @@ abstract class CoreMainActivity : BaseActivity(), WebViewProvider {
     return supportFragmentManager.fragments.filterIsInstance<WebViewProvider>().firstOrNull()
       ?.getCurrentWebView()
   }
+
+  abstract fun setupDrawerToggle(toolbar: Toolbar)
+
+  override fun onNavigationItemSelected(item: MenuItem): Boolean {
+    when (item.itemId) {
+      R.id.menu_support_kiwix -> openSupportKiwixExternalLink()
+      R.id.menu_settings -> openSettingsActivity()
+      R.id.menu_help -> start<HelpActivity>()
+      R.id.menu_history -> openHistoryActivity()
+      R.id.menu_bookmarks_list -> openBookmarksActivity()
+      else -> return false
+    }
+    return true
+  }
+
+  override fun onBackPressed() {
+    if (navigationDrawerIsOpen()) {
+      closeNavigationDrawer()
+      return
+    }
+    supportFragmentManager.fragments.filterIsInstance<BaseFragmentActivityExtensions>().forEach {
+      if (it.onBackPressed(this) == BaseFragmentActivityExtensions.Super.ShouldCall) {
+        super.onBackPressed()
+      }
+    }
+  }
+
+  abstract fun navigationDrawerIsOpen(): Boolean
+  abstract fun closeNavigationDrawer()
+
+  private fun openSupportKiwixExternalLink() {
+    openExternalUrl(
+      sharedPreferenceUtil,
+      alertDialogShower,
+      Uri.parse("https://www.kiwix.org/support").browserIntent().putExtra(
```

Comment:
I believe we have the extension function from `androidx-core` `.toUri` which is just a wrapper around `Uri.parse`

Reply:
also the string here should be a constant